core/roboconf-plugin-script/src/main/java/net/roboconf/plugin/script/internal/PluginScript.java

```diff
@@ -167,16 +168,25 @@ public void undeploy( Instance instance ) throws PluginException {
     private void prepareAndExecuteCommand(String action, Instance instance, Import importChanged, InstanceStatus statusChanged)
     throws IOException, InterruptedException {
 
-        this.logger.info("Preparing the invocation of " + action + ".sh for instance " + instance );
+        this.logger.info("Preparing the invocation of " + action + " script for instance " + instance );
         File instanceDirectory = InstanceHelpers.findInstanceDirectoryOnAgent( instance );
 
         File scriptsFolder = new File(instanceDirectory, SCRIPTS_FOLDER_NAME);
         File templatesFolder = new File(instanceDirectory, TEMPLATES_FOLDER_NAME);
 
+        // Look for action script (default <action>.sh, or any file that starts with <action>)
         File script = new File(scriptsFolder, action + ".sh");
-        File template = new File(templatesFolder, action + ".sh.template");
+        if(! script.exists()) {
+        	File[] foundFiles = scriptsFolder.listFiles(new ActionFileFilter(action));
```

Comment:
Just done... by changing test scripts names (using several extensions - like .bash or .whatever or even nothing - instead of ".sh"). See src/test/resources/BashScript/script .